app/api/ddi-index-api/users.js

```diff
@@ -19,21 +21,62 @@ const usersEndpoint = 'users'
  */
 
 /**
+ * @typedef UserAccount
+ * @property {number} id
+ * @property {string} username
+ * @property {number} policeForceId
+ * @property {string} policeForce
+ * @property {boolean|Date} accepted
+ * @property {boolean|Date} activated
+ * @property {boolean|Date} lastLogin
+ * @property {boolean|Date} createdAt
+ */
+
+/**
+ * @typedef GetUserOptions
+ * @property {{ policeForceId?: number }} [filter]
+ * @property {{
+ *    username?: 'ASC'|'DESC';
+ *    policeForce?: 'ASC'|'DESC';
+ *    indexAccess?: boolean;
+ * }} [sort]
+ */
+
+/**
+ * @param {GetUserOptions} options
  * @param callingUser
- * @return {Promise<{ users: UserAccount[], count: number }>}
+ * @return {Promise<{ users: UserAccount[]; count: number }>}
  */
-const getUsers = async (callingUser) => {
+const getUsers = async (options, callingUser) => {
+  const url = new URL(`${usersEndpoint}`, config.ddiIndexApi.baseUrl)
+
+  if (!isNaN(options.filter?.policeForceId)) {
+    url.searchParams.append('policeForceId', `${options.filter.policeForceId}`)
+  }
+
+  let sortOrder
+
+  if (options.sort?.username !== undefined) {
+    url.searchParams.append('sortKey', 'username')
+    sortOrder = options.sort.username
+  } else if (options.sort?.policeForce !== undefined) {
+    url.searchParams.append('sortKey', 'policeForce')
+    sortOrder = options.sort.policeForce
+  } else if (options.sort?.indexAccess !== undefined) {
+    url.searchParams.append('sortKey', 'activated')
+    url.searchParams.append('activated', `${options.sort.indexAccess}`)
+  }
+
+  if ([sort.ASC, sort.DESC].includes(sortOrder)) {
+    url.searchParams.append('sortOrder', sortOrder)
+  }
```

Comment:
Do you need to adda default sort order if not supplied, or is that defaulted by the API?

Reply:
the returns the results sorted by username by default